src/Navigation/WPF/Navigation.WPF/Service/NavigationService.cache.cs

```diff
@@ -21,25 +21,56 @@ public void ClearCache()
             ContentCache.Clear();
         }
 
-        private void CreateCachedContent(
-            Type viewType)
+
+        private static void CreateCachedContent(
+            INavigationRegistration registration)
         {
             if (!Configuration.CacheContent)
             {
                 return;
             }
 
             if (Configuration.ContentTypesToExcludeFromCaching
-                .Contains(viewType))
+                .Contains(registration.ViewType))
             {
                 return;
             }
 
-            var content = (FrameworkElement) Activator.CreateInstance(
-                viewType);
+            var content = GetContentInstance<FrameworkElement>(
+                registration);
 
             var cachedContent = new CachedContent(content);
             ContentCache.Add(cachedContent);
         }
+
+
+        private static TContent AddOrUpdateContentCache<TContent>(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused private method 'AddOrUpdateContentCache'.](https://app.codacy.com/gh/UltimateCodeMonkeys/CodeMonkeysMVVM/pullRequest?prid=5575746)